store/src/main/java/com/dropbox/android/external/store4/SourceOfTruth.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dropbox.android.external.store4
+
+import com.dropbox.android.external.store4.impl.PersistentNonFlowingSourceOfTruth
+import com.dropbox.android.external.store4.impl.PersistentSourceOfTruth
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * Source of truth takes care of making any source (no matter if it has flowing reads or not) into
```

Comment:
this may not be very clear for someone new as it does not explain what SOT does, rather explains what edge cases it covers.

Maybe something like this:

```
SourceOfTruth, as name implies, is the persistence API which Store uses to serve values to the collectors. If provided, Store will only return values received from `SourceOfTruth` back to the collectors. In other words, values coming from the Fetcher will always be sent to the `SourceOfTruth` and will be read back via [reader] to be returned to the collector. This round-trip ensures the data is consistent across the application in case the `Fetcher` may not return all fields or return a different class type than the app uses.

It is particularly useful if your application has a local observable database which is directly modified by the app as Store can observe these changes and update the collectors even before value is synced to the backend.
```

Reply:
ahaha funny i just wrote a similar comment on the RX one since it was shown above :)